sql/core/src/test/scala/org/apache/spark/sql/StatisticsCollectionSuite.scala

```diff
@@ -678,4 +680,50 @@ class StatisticsCollectionSuite extends StatisticsCollectionTestBase with Shared
       }
     }
   }
+
+  test("SPARK-34137: Update suquery's stats when build LogicalPlan's stats") {
+    withTable("t1", "t2") {
+      sql("create table t1 using parquet as select id as a, id as b from range(1000)")
+      sql("create table t2 using parquet as select id as c, id as d from range(2000)")
+
+      sql("ANALYZE TABLE t1 COMPUTE STATISTICS FOR ALL COLUMNS")
+      sql("ANALYZE TABLE t2 COMPUTE STATISTICS FOR ALL COLUMNS")
+      sql("set spark.sql.cbo.enabled=true")
+
+      val df = sql(
+        """
+          |WITH max_store_sales AS
+          |(
+          |  SELECT max(csales) tpcds_cmax
+          |  FROM (
+          |    SELECT sum(b) csales
+          |    FROM t1 WHERE a < 100
+          |  ) x
+          |),
+          |best_ss_customer AS
+          |(
+          |  SELECT c
+          |  FROM t2
+          |  WHERE d > (SELECT * FROM max_store_sales)
+          |)
+          |SELECT c FROM best_ss_customer
+          |""".stripMargin)
+      val optimizedPlan = df.queryExecution.optimizedPlan
+      optimizedPlan.stats
+      val subqueryExpression = ArrayBuffer.empty[SubqueryExpression]
```

Comment:
nit: `mutable.ArrayBuffer.empty[SubqueryExpression]`

(since we already imported `mutable` package)

Reply:
> nit: `mutable.ArrayBuffer.empty[SubqueryExpression]`
> 
> (since we already imported `mutable` package)

Updated